inc/targetinterface.class.php

```diff
@@ -37,5 +37,27 @@ interface PluginFormcreatorTargetInterface
 {
    public function save(PluginFormcreatorFormAnswer $formanswer);
 
-   public function getTargetItemtypeName();
+   public static function getTargetItemtypeName();
+
+   /**
+    * Tell if this target does an action and does not generate an object
+    *
+    * @return boolean true if the target does an action
+    */
+   public function isActionTarget(): bool;
+
+   /**
+    * Tell if this target generates an object and does not does an action
+    *
+    * @return boolean true if the target generates an object
+    */
+   public function isObjectTarget(): bool;
```

Comment:
The comments seems to indicate that `isActionTarget` and `isObjectTarget` are exclusive.
Wouldn't a single method `getTargetType` returning something like `AbstractTarget::TYPE_ACTION` or `AbstractTarget::TYPE_OBJECT` be better ?

`isActionTarget` and `isObjectTarget` could stays as helper methods in the base abstract class but only `getTargetType` would be required by the interface.

Reply:
> The comments seems to indicate that isActionTarget and isObjectTarget are exclusive

Yes I consider they are exclusive. Actually creating a ticket is already an action by itself, but it creates an object on the DB. Your SentEmail is also an object, then EmailTarget generates an object. 

> The comments seems to indicate that isActionTarget and isObjectTarget are exclusive.
Wouldn't a single method getTargetType returning something like AbstractTarget::TYPE_ACTION or AbstractTarget::TYPE_OBJECT be better ?
>
> **isActionTarget** and **isObjectTarget** could stays as helper methods in the base abstract class but only getTargetType would be required by the interface.

If I switch to a single method returning a constant, then I wwould drop **isActionTarget** and **isObjectTarget**  from this PR.